posthog/src/main/java/com/posthog/PostHogOkHttpInterceptor.kt

```diff
@@ -0,0 +1,52 @@
+package com.posthog
+
+import com.posthog.internal.RRPluginEvent
+import com.posthog.internal.capture
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+
+public class PostHogOkHttpInterceptor(private val captureNetwork: Boolean = true) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val originalRequest = chain.request()
+
+        try {
+            val response = chain.proceed(originalRequest)
+
+            captureNetworkEvent(originalRequest, response)
+
+            return response
+        } catch (e: Throwable) {
+            throw e
+        }
+    }
+
+    private fun captureNetworkEvent(request: Request, response: Response) {
+        if (!captureNetwork) {
+            return
+        }
+        val url = request.url.toString()
+        val method = request.method
+        val statusCode = response.code
+        val start = response.sentRequestAtMillis
+        val end = response.receivedResponseAtMillis
+        var transferSize = (response.body?.contentLength() ?: 0) + (request.body?.contentLength() ?: 0)
+
+        if (transferSize < 0) {
+            transferSize = 0
+        }
```

Comment:
ooh, this is cool... `Response` has nullable `cacheResponse()` and `networkResponse()` so you can distinguish between cached and non-cached responses. Which would let you decide when to return `transferSize` = 0

We could have `responseFromCache` on the payload and defuckulate that in the UI too for more clarity. I don't want to pedantically stick to the spec just not vary for the fun of it :)

Glad that this has headers and body available for when we want to start letting people capture them... that's going to be 💸